notify_audit: false
locked: false
name: '[${environment}] external-dns reporting errors registering records'
tags: [network, '${environment}']
include_tags: false
no_data_timeframe: null
silenced: {}
new_host_delay: 300
require_full_window: false
notify_no_data: false
renotify_interval: 0
escalation_message: ''
query: sum(last_15m):max:external_dns.registry_errors_total{${cluster_tag}:${cluster}}.as_count() > 1000
message: |
  {{#is_warning}}
  The external-dns controller is reporting registry errors.
  - Is there a conflicting record in the registry (e.g. an existing CNAME)?
  - Does external-dns have proper permissions to access the registry?

  Check the logs of external-dns for more information.
  ${low_urgency_notifications}
  {{/is_warning}}
  {{#is_recovery}}
  ${low_urgency_notifications}
  {{/is_recovery}}
type: metric alert
thresholds:
  critical: 1000
  warning: ${external_dns_registry_errors_warning_threshold}
timeout_h: 0
definition_defaults:
  external_dns_registry_errors_warning_threshold: 1
  cluster_tag: kubernetescluster
